caps/eth.md

```diff
@@ -0,0 +1,223 @@
+Ethereum blockchain information between peers. The current protocol version is **eth/63**.
+See end of document for a list of changes in past protocol versions.
+
+### Protocol Semantics
+
+Upon an active session, a Status message must be sent. Following the reception of the
+peer's Status message, the Ethereum session is active and any other messages may be sent.
+
+All known transactions should be sent following the Status exchange with one or more
+Transactions messages.
+
+Transactions messages should also be sent periodically as the node has new transactions to
+disseminate. A node should never send a transaction back to the peer that it can determine
+already knows of it (either because it was previously sent or because it was informed from
+this peer originally).
+
+New blocks are propagated using the NewBlock and NewBlockHashes messages. The NewBlock
+message includes the full block and is sent to a small fraction of connected peers
+(usually sqrt(peers)). All other peers are sent a NewBlockHashes message containing just
+the hash of the new block. Those peers may request the full block later if they fail to
+receive it from anyone within reasonable time.
+
+Blocks are typically re-propagated to all connected peers as soon as basic validity of the
+announcement has been established (e.g. after the proof-of-work check).
+
+### Basic Chain Synchronization
+
+Two peers connect and send their Status message. Status includes the Total Difficulty(TD)
+and hash of their best block.
+
+The client with the worst TD then proceeds to download block headers using the
+BlockHeadersFromNumber message. It verifies proof-of-work values in received headers and
+fetches block bodies using the GetBlockBodies message. Received blocks are executed using
+the Ethereum Virtual Machine, recreating the state tree.
+
+Note that header downloads, block body downloads and block execution may happen
+concurrently.
+
+### State Synchronization (a.k.a. "fast sync")
+
+eth/63 also allows synchronizing transaction execution results ("state"). This may be
+faster than re-executing all transactions but comes at the expense of some security.
+
+State synchronization typically proceeds by downloading the chain of block headers,
+verifying their proof-of-work values. Block bodies are requested as in the Basic Chain
+Synchronization section but block transactions aren't executed. Instead, the client picks
+a block near the head of the chain and downloads merkle tree nodes and contract code
+incrementally by requesting the root node, it's children, grandchildren, ... until the
+entire tree is synchronized.
+
+## Protocol Messages
+
+### Status (0x00)
+
+`[protocolVersion: P, networkId: P, td: P, bestHash: B_32, genesisHash: B_32, number: P]`
+
+Inform a peer of its current ethereum state. This message should be sent _after_ the
+initial handshake and _prior_ to any **ethereum** related messages.
```

Comment:
Again I think if the 'initial handshake' is referring to the RLPx transport handshake, it should not really be part of this protocol. Or rather, I do remember being confused reading the original document a year ago, trying to establish if this handshake they were referring to was part of this protocol or some other. Maybe this could be rephrased to say "after the underlying transport has completed the handshake and the connection is established, the first message of this Eth protocol must be...."

Reply:
I don't want to refer to RLPx specifics in this document.